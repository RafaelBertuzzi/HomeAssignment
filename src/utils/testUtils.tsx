import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import { LayoutProvider } from '@components/LayoutProvider';
import { NavigationContainer } from '@react-navigation/native';
import { store } from '@redux/store';
import { render } from '@testing-library/react-native';

export const rendererWithProviders = (children: ReactNode) => {
  return renderer
    .create(
      <Provider store={store}>
        <NavigationContainer>
          <LayoutProvider>{children}</LayoutProvider>
        </NavigationContainer>
      </Provider>,
    )
    .toJSON();
};

export const renderWithProviders = (children: ReactNode) => {
  return render(
    <Provider store={store}>
      <NavigationContainer>
        <LayoutProvider>{children}</LayoutProvider>
      </NavigationContainer>
    </Provider>,
  ).toJSON();
};
