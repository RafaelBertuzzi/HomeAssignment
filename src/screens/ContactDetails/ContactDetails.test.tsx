import { Contact } from 'react-native-contacts';

import { cleanup } from '@testing-library/react-native';
import { rendererWithProviders } from '@utils/testUtils';

import { ContactDetails } from '.';

const mockedData = {
  birthday: { day: 20, month: 1, year: 1978 },
  company: 'Creative Consulting',
  emailAddresses: [],
  familyName: 'Bell',
  givenName: 'Kate',
  hasThumbnail: false,
  imAddresses: [],
  jobTitle: 'Producer',
  middleName: '',
  phoneNumbers: [],
  postalAddresses: [],
  recordID: '177C371E-701D-42F8-A03B-C61CA31627F6',
  thumbnailPath: '',
  urlAddresses: [],
  backTitle: 'Back',
  department: 'Marketing',
  displayName: 'Kate Bell',
  isStarred: false,
  note: 'Kate Bell has been working on new designs for our holiday product line. She has some great ideas.',
  prefix: 'Ms.',
  suffix: 'Jr.',
} as Contact;

jest.mock('@features/ContactDetails/hooks/useContactDetails', () => ({
  useContactDetails: () => ({
    data: mockedData,
    isLoading: false,
  }),
}));

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useRoute: () => ({
    params: { id: mockedData.recordID },
  }),
}));

describe('ContactDetails Screen', () => {
  afterEach(cleanup);

  it('should render ContactDetails Screen', () => {
    const json = rendererWithProviders(<ContactDetails />);
    expect(json).toMatchSnapshot();
  });
});
