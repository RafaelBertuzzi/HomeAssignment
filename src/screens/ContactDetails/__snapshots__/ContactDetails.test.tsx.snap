// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetails Screen should render ContactDetails Screen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    dataSet={{}}
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "position": "fixed",
        }
      }
    >
      <View
        dataSet={{}}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 16,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          colorScheme="primary"
          dataSet={{}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 10,
            }
          }
        >
          <View
            dataSet={{}}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
            test={true}
          >
            <View
              dataSet={{}}
              style={{}}
            >
              <Image
                accessibilityLabel="left-arrow"
                alt="left-arrow"
                dataSet={{}}
                onError={[Function]}
                source={
                  {
                    "testUri": "../../../src/assets/icons/left-arrow.png",
                  }
                }
                style={
                  {
                    "height": 24,
                    "maxWidth": "100%",
                    "width": 24,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          dataSet={{}}
          style={{}}
        />
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          colorScheme="primary"
          dataSet={{}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 10,
            }
          }
        >
          <View
            dataSet={{}}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
            test={true}
          >
            <View
              dataSet={{}}
              style={{}}
            >
              <Image
                accessibilityLabel="star-outline"
                alt="star-outline"
                dataSet={{}}
                onError={[Function]}
                source={
                  {
                    "testUri": "../../../src/assets/icons/star-outline.png",
                  }
                }
                style={
                  {
                    "height": 24,
                    "maxWidth": "100%",
                    "width": 24,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        aria-orientation="horizontal"
        dataSet={{}}
        style={
          {
            "backgroundColor": "#d4d4d4",
            "height": 1,
            "width": "100%",
          }
        }
        thickness="1"
      />
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <View
      dataSet={{}}
      style={
        {
          "flex": 1,
          "marginBottom": 16,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 16,
        }
      }
    >
      <RCTScrollView>
        <View>
          <View
            dataSet={{}}
            style={
              {
                "flex": 1,
                "width": "100%",
              }
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Name:
              </Text>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 16,
                  }
                }
              />
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Kate Bell 
              </Text>
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "height": 16,
                }
              }
            />
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Birthday:
              </Text>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 16,
                  }
                }
              />
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                20/01/1978
              </Text>
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "height": 16,
                }
              }
            />
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Phone Numbers:
              </Text>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 16,
                  }
                }
              />
              <View
                dataSet={{}}
                style={
                  {
                    "width": "50%",
                  }
                }
              />
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "height": 16,
                }
              }
            />
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Email Addresses:
              </Text>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 16,
                  }
                }
              />
              <View
                dataSet={{}}
                style={
                  {
                    "width": "50%",
                  }
                }
              />
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "height": 16,
                }
              }
            />
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                URL Addresses:
              </Text>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 16,
                  }
                }
              />
              <View
                dataSet={{}}
                style={
                  {
                    "width": "50%",
                  }
                }
              />
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "height": 16,
                }
              }
            />
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Job Title:
              </Text>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 16,
                  }
                }
              />
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Producer
              </Text>
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "height": 16,
                }
              }
            />
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Company:
              </Text>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 16,
                  }
                }
              />
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Creative Consulting
              </Text>
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "height": 16,
                }
              }
            />
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Postal Addresses:
              </Text>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 16,
                  }
                }
              />
              <View
                dataSet={{}}
                style={
                  {
                    "width": "50%",
                  }
                }
              />
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
